### PR TITLE
Working client subscriptions callbacks

### PR DESCRIPTION
**Pull request by sergioherrera7**

### CODE REVIEW

File: include/open62541cpp/open62541client.h [hunk 1]
```diff
@@ -241,14 +241,24 @@ class Client
 
     /*!
      * \brief Initialise custom client config
+     * \param clientConfig client configuration
+     * \param setDefaultConfig flag to reset client config context
      */
-    void setCustomConfig(UA_ClientConfig* clientConfig)
+    void setCustomConfig(UA_ClientConfig* clientConfig, bool setDefaultConfig = true)
```
Comment (sergioherrera7) [inline include/open62541cpp/open62541client.h:247]: This is the only change that matters. It is non-destructive so merging it should not break anything.

File: CMakeLists.txt [hunk 2]
```diff
@@ -11,7 +11,7 @@ set(OPEN62541_CPP open62541cpp)
 ## build settings - try and match the c library version
 set(VERSION_MAJOR 1)
 set(VERSION_MINOR 2)
-set(VERSION_PATCH 666)
+set(VERSION_PATCH 777)
```
Comment (sergioherrera7) [inline CMakeLists.txt:14]: autocrane_core has to be updated to fetch this version

File: examples/TestClientDefaultEncryption/main.cpp [hunk 3]
```diff
@@ -151,8 +165,13 @@ int main()
...
      * with the Uri specified below.*/
     UA_ApplicationDescription_clear(&clientConfig->clientDescription);
-    clientConfig->clientDescription.applicationUri  = UA_STRING_ALLOC("urn:freeopcua:client");
+    clientConfig->clientDescription.applicationUri  = UA_String_fromChars("urn:freeopcua:client");
+    // clientConfig->applicationUri  = UA_String_fromChars("urn:freeopcua:client");
     clientConfig->clientDescription.applicationType = UA_APPLICATIONTYPE_CLIENT;
+    // clientConfig->outStandingPublishRequests = 0;
+
+    cout << "setting CustomConfig..." << endl << endl;
+    client.setCustomConfig(clientConfig,false);
```
Comment (axelperschmann) [inline examples/TestClientDefaultEncryption/main.cpp]:
```suggestion
    client.setCustomConfig(clientConfig, false);
```

File: examples/TestClientDefaultEncryption/main.cpp [hunk 4]
```diff
@@ -115,33 +117,45 @@ int main()
...
+    // printf("Reading server_cert=%s \n", server_cert.c_str());
     UA_ByteString certificate = loadFile(certfile);
     UA_ByteString privateKey  = loadFile(keyfile);
+    // UA_ByteString server_certificate  = loadFile(server_cert);
+
+    // UA_ByteString trustList[1];
+    // trustList[0] = server_certificate;
+    // size_t trustListSize = sizeof(trustList) / sizeof(UA_ByteString);
+    // printf("trustList[0].length=%zu \n", trustList[0].length);
+    // printf("trustListSize=%zu \n", trustListSize);
```
Comment (eilers) [inline examples/TestClientDefaultEncryption/main.cpp]: Please remove code that was commented out.

File: examples/TestClientDefaultEncryption/main.cpp [hunk 5]
```diff
@@ -115,33 +117,45 @@ int main()
     // clientConfig->logger = customLogger;
     cout << "Redirecting output stream...done" << endl;
 
-
     // cout << "setCustomConfig..." << endl << endl;
     // client.setCustomConfig(clientConfig);
```
Comment (axelperschmann) [inline examples/TestClientDefaultEncryption/main.cpp]: remove commented out code, or add now existing param `setCustomConfig` so one does not accidentially comment in method call with default param?

File: CMakeLists.txt (hunk 2 above)
Comment (ahoereth) [reply CMakeLists.txt:14]: and it should be tagged once this pr is merged

**Comment by sergioherrera7** [comment]
Will merge so we can tag the commit and use the id on: https://github.com/psiori/autocrane-core/pull/3224

File: CMakeLists.txt (hunk 2 above)
Comment (sergioherrera7) [reply CMakeLists.txt:14]: tagged as v1.2.77 ->  https://github.com/psiori/open62541Cpp/releases/tag/v1.2.777

**Pull request merged**